packages/flutter/lib/src/widgets/localizations.dart

```diff
@@ -0,0 +1,465 @@
+    // https://github.com/dart-lang/sdk/issues/27680 has been resolved.
+    return resources[type];
+  }
+
+  @override
+  bool shouldReload(DefaultLocalizationsDelegate old) => false;
+}
+
+class _MergedLocalizationsDelegate extends LocalizationsDelegate {
+  // Creates a single [LocalizationsDelegate] whose methods delegate to the
+  // elements of `allDelegates`.
+  _MergedLocalizationsDelegate(this.allDelegates);
+
+  // This class's [load] and [resourcesFor] methods delegate
+  // to the members of this list.
+  final Iterable<LocalizationsDelegate> allDelegates;
+
+  final Map<Locale, Map<Type, LocalizationsDelegate>> _localeToDelegate = <Locale, Map<Type, LocalizationsDelegate>>{};
+  final Set<Locale> _loading = new Set<Locale>();
+
+  @override
+  Future<Iterable<Type>> load(Locale locale) {
+    assert(locale != null);
+    assert(!_loading.contains(locale));
+
+    List<Type> types;
+
+    _loading.add(locale);
+    final Future<Iterable<Type>> typesFuture = _loadMap<LocalizationsDelegate, Iterable<Type>>(
+      new Map<LocalizationsDelegate, Future<Iterable<Type>>>.fromIterables(
+        allDelegates,
+        allDelegates.map<Future<Iterable<Type>>>((LocalizationsDelegate delegate) => delegate.load(locale)),
+      ),
+    ).then<Iterable<Type>>((Map<LocalizationsDelegate, Iterable<Type>> resources) {
+      _loading.remove(locale);
+      final Map<Type, LocalizationsDelegate> typeToDelegate = <Type, LocalizationsDelegate>{};
+      for (LocalizationsDelegate delegate in resources.keys) {
+        for (Type type in resources[delegate]) {
+          assert(type != null && !typeToDelegate.containsKey(type));
+          typeToDelegate[type] = delegate;
+          types ??= <Type>[];
+          types.add(type);
+        }
+      }
+      _localeToDelegate[locale] = typeToDelegate;
+      return types;
+    });
+
+    return types == null ? typesFuture : new SynchronousFuture<Iterable<Type>>(types);
+  }
+
+  @override
+  T resourcesFor<T>(Locale locale, Type type) {
+    assert(locale != null);
+    assert(type != null);
+    final Map<Type, LocalizationsDelegate> resources = _localeToDelegate[locale];
+    return resources[type]?.resourcesFor<T>(locale, type);
+  }
+
+  @override
+  bool shouldReload(_MergedLocalizationsDelegate old) => false;
+}
+
+class _LocalizationsScope extends InheritedWidget {
+  _LocalizationsScope ({
+    Key key,
+    @required this.locale,
+    @required this.localizationsState,
+    Widget child,
+  }) : super(key: key, child: child) {
+    assert(localizationsState != null);
+  }
+
+  final Locale locale;
+  final _LocalizationsState localizationsState;
+
+  @override
+  bool updateShouldNotify(_LocalizationsScope old) {
+    final LocalizationsDelegate delegate = localizationsState.widget.delegate;
+    final bool shouldReload = locale != old.locale ||
+      (delegate != null && delegate.shouldReload(old.localizationsState.widget.delegate));
```

Comment:
localizationsState.widget.delegate and old.localizationsState.widget.delegate will always be the same delegate, because localizationsState and old.localizationsState will always be the same